From b7aa24c9c82eabda10b9a5d8968448e52c1da04f Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 17 May 2016 09:12:56 +0300
Subject: [PATCH 476/538] vbus: a7040: add regulator node for USB-Host 0/1 VBUS

Add regulator node for USB-Host 0/1 VBUS using IO-Expander
Using I2C device 0x21, offset 0/1 for USB Host 0/1.

Change-Id: Ib971b5bf7af648389388d1cffea1f528de314aa7
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30106
---
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi | 43 +++++++++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-7040.dtsi    |  6 ++++
 2 files changed, 49 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index 97f3ff5..404c2e7 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -75,6 +75,23 @@
 
 	cpn-110-master {
 		config-space {
+			i2c1: i2c@701000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&i2c0_pins>;
+				expander0: pca9555@21 {
+					compatible = "nxp,pca9555";
+					pinctrl-names = "default";
+					gpio-controller;
+					#gpio-cells = <2>;
+					reg = <0x21>;
+				};
+			};
+			usb3h0: usb3@500000 {
+				usb-phy = <&usb3h0_phy>;
+			};
+			usb3h1: usb3@510000 {
+				usb-phy = <&usb3h1_phy>;
+			};
 			spi@700680 {
 				spi-flash@0 {
 					#address-cells = <0x1>;
@@ -114,4 +131,30 @@
 			};
 		};
 	};
+	usb3h0_phy: usb3-h0-phy {
+		compatible = "usb-nop-xceiv";
+		vcc-supply = <&exp_usb3h0_vbus>;
+	};
+	usb3h1_phy: usb3-h1-phy {
+		compatible = "usb-nop-xceiv";
+		vcc-supply = <&exp_usb3h1_vbus>;
+	};
+	exp_usb3h0_vbus: usb3-0-vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "usb3-0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		regulator-always-on;
+		gpio = <&expander0 0 GPIO_ACTIVE_HIGH>;
+	};
+	exp_usb3h1_vbus: usb3-1-vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "usb3-1-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		regulator-always-on;
+		gpio = <&expander0 1 GPIO_ACTIVE_HIGH>;
+	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index 6e14bf4..f73472b 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -68,6 +68,12 @@
 
 	cpn-110-master {
 		config-space {
+			pinctrl@440000 {
+				i2c0_pins: i2c-pins-0 {
+					marvell,pins = "mpp37", "mpp38";
+					marvell,function = "i2c0";
+				};
+			};
 			gpio1: gpio@440100 {
 			};
 			gpio2: gpio@440140 {
-- 
1.9.1

